core/util/__init__.py

```diff
@@ -526,15 +528,21 @@ class MoneyUtility:
     DEFAULT_CURRENCY = "USD"
 
     @classmethod
-    def parse(cls, amount):
+    def parse(cls, amount: str | float | int | None) -> Money:
         """Attempt to turn a string into a Money object."""
         currency = cls.DEFAULT_CURRENCY
         if not amount:
             amount = "0"
         amount = str(amount)
+
         if amount[0] == "$":
             currency = "USD"
             amount = amount[1:]
+
+        # `Money` does not properly handle commas in the amount, so we strip
+        # them out of US dollar amounts, since it won't change the "value."
+        if currency == "USD":
+            amount = "".join(amount.split(","))
```

Comment:
This does the job but, I wonder though if we would be better off using something like the Babel number parser to create this value?

https://babel.pocoo.org/en/latest/api/numbers.html#babel.numbers.parse_decimal

Might handle some more of the edge cases here.

Reply:
It  looked at it, but it seemed to be more trouble than it was worth. Its strict mode doesn't allow extra zeros after the decimal point (e.g., "4,444.4" is valid, but "4,444.40" is not) and we'd need to handle its exception or map it into a `ValueError`.

Reply:
Thats annoying! I thought Babel would have handled those cases better then that. I'll mark this as resolved then!

Reply:
Yeah, it did seem kind of dumb, but that's why I gave it pass.